src/lib/pubkey/classic_mceliece/cmce_poly.cpp

```diff
@@ -0,0 +1,175 @@
+   return create_element_from_coef(coef);
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::create_element_from_coef(
+   const std::vector<GF_Elem>& coeff_vec) const {
+   std::vector<Classic_McEliece_GF> coeff_vec_gf;
+   std::transform(coeff_vec.begin(), coeff_vec.end(), std::back_inserter(coeff_vec_gf), [this](auto& coeff) {
+      return Classic_McEliece_GF(coeff, m_poly_f);
+   });
+   return Classic_McEliece_Polynomial(coeff_vec_gf);
+}
+
+bool operator==(const Classic_McEliece_Polynomial_Ring::Big_F_Coefficient& lhs,
+                const Classic_McEliece_Polynomial_Ring::Big_F_Coefficient& rhs) {
+   return lhs.coeff == rhs.coeff && lhs.idx == rhs.idx;
+}
+
+std::optional<Classic_McEliece_Minimal_Polynomial> Classic_McEliece_Polynomial_Ring::compute_minimal_polynomial(
+   std::span<const uint8_t> seed) const {
+   auto polynomial = create_element_from_bytes(seed);
+   std::vector<Classic_McEliece_Polynomial> mat;
+
+   mat.push_back(create_element_from_coef(concat_as<std::vector<GF_Elem>>(
+      std::vector<GF_Elem>{GF_Elem(1)}, std::vector<GF_Elem>(degree() - 1, GF_Elem(0)))));
+
+   mat.emplace_back(polynomial);
+
+   for(size_t j = 2; j <= degree(); ++j) {
+      mat.push_back(multiply(mat.at(j - 1), polynomial));
+   }
+
+   // Gaussian
+   for(size_t j = 0; j < degree(); ++j) {
+      for(size_t k = j + 1; k < degree(); ++k) {
+         auto cond = GF_Mask::is_zero(mat.at(j).coef_at(j));
+
+         for(size_t c = j; c < degree() + 1; ++c) {
+            mat.at(c).coef_at(j) += cond.if_set_return(mat.at(c).coef_at(k));
+         }
+      }
+
+      if(mat.at(j).coef_at(j).is_zero()) {
+         // Fail if not systematic.
+         return std::nullopt;
+      }
+
+      auto inv = mat.at(j).coef_at(j).inv();
+
+      for(size_t c = j; c < degree() + 1; ++c) {
+         mat.at(c).coef_at(j) *= inv;
+      }
+
+      for(size_t k = 0; k < degree(); ++k) {
+         if(k != j) {
+            auto t = mat.at(j).coef_at(k);
+
+            for(size_t c = j; c < degree() + 1; ++c) {
+               mat.at(c).coef_at(k) += mat.at(c).coef_at(j) * t;
+            }
+         }
+      }
+   }
+
+   auto minimal_poly_coeffs = mat.at(degree()).coef();
+   // Add coefficient 1 since polynomial is monic
+   minimal_poly_coeffs.emplace_back(GF_Elem(1), poly_f());
+
+   return Classic_McEliece_Minimal_Polynomial(std::move(minimal_poly_coeffs));
+}
+
+secure_vector<uint8_t> Classic_McEliece_Minimal_Polynomial::serialize() const {
+   BOTAN_ASSERT_NOMSG(!coef().empty());
+   auto& all_coeffs = coef();
+   // Store all except coef for monomial x^t since polynomial is monic (ISO Spec Section 9.2.9)
+   auto coeffs_to_store = std::span(all_coeffs).subspan(0, all_coeffs.size() - 1);
+   secure_vector<uint8_t> bytes(sizeof(uint16_t) * coeffs_to_store.size());
+   BufferStuffer bytes_stuf(bytes);
+   for(auto& coef : coeffs_to_store) {
+      store_le(coef.elem().get(), bytes_stuf.next(sizeof(GF_Elem)).data());
```

Comment:
```suggestion
      store_le(bytes_stuf.next<sizeof(GF_Elem)>(), coef.elem().get());
```

This will create a statically-sized span from the stuffer, that can be bounds-checked at compile time.